### ISSUE TITLE
Pebble `replan`, restart, and any other invocation does not seem to re-invoke with new env variables

### ISSUE THREAD

**Issue by rbarry82** [canonical/pebble#102]
Grafana is commonly provisioned using environment variables to override defaults rather than using the config file. It doesn't support `SIGHUP` in any case.

When the environment for a service is changed, the new Pebble plan accurately reflects it. When the service is restarted (or replanned), it's restarted as normal.

*But* the new environment variables are not present.

After a new config value was set in the charm which updated the environment:

```
root@grafana-0:/usr/share/grafana# /charm/bin/pebble plan                                                                                                                                                                                       
services:                                                                                                                                                                                                                                       
    grafana:                                                                                                                                                                                                                                    
        summary: grafana-k8s service                                                                                                                                                                                                            
        startup: enabled                                                                                                                                                                                                                        
        override: replace
        command: grafana-server -config /etc/grafana/grafana-config.ini
        environment:
            GF_DATABASE_TYPE: sqlite3
            GF_LOG_LEVEL: info
            GF_PATHS_PROVISIONING: /etc/grafana/provisioning 
            GF_SECURITY_ADMIN_PASSWORD: rIMaoW7Fml26
            GF_SECURITY_ADMIN_USER: admin
            GF_SERVER_HTTP_PORT: "3000"
            GF_SERVER_ROOT_URL: http://0.0.0.0:3000/grafana
            GF_SERVER_SERVE_FROM_SUB_PATH: "true"
root@grafana-0:/usr/share/grafana# xargs -0 -L1 -a /proc/45/environ | sort
GF_DATABASE_TYPE=sqlite3
GF_LOG_LEVEL=info
GF_PATHS_CONFIG=/etc/grafana/grafana.ini
GF_PATHS_DATA=/var/lib/grafana
GF_PATHS_HOME=/usr/share/grafana
GF_PATHS_LOGS=/var/log/grafana
GF_PATHS_PLUGINS=/var/lib/grafana/plugins
GF_PATHS_PROVISIONING=/etc/grafana/provisioning
GF_SECURITY_ADMIN_PASSWORD=rIMaoW7Fml26
GF_SECURITY_ADMIN_USER=admin
GF_SERVER_HTTP_PORT=3000
GRAFANA_PORT=tcp://10.152.183.68:65535
GRAFANA_PORT_65535_TCP=tcp://10.152.183.68:65535
GRAFANA_PORT_65535_TCP_ADDR=10.152.183.68
GRAFANA_PORT_65535_TCP_PORT=65535
GRAFANA_PORT_65535_TCP_PROTO=tcp
GRAFANA_SERVICE_HOST=10.152.183.68
GRAFANA_SERVICE_PORT=65535
GRAFANA_SERVICE_PORT_PLACEHOLDER=65535
HOME=/root
HOSTNAME=grafana-0
JUJU_CONTAINER_NAME=grafana
KUBERNETES_PORT=tcp://10.152.183.1:443
KUBERNETES_PORT_443_TCP=tcp://10.152.183.1:443
KUBERNETES_PORT_443_TCP_ADDR=10.152.183.1
KUBERNETES_PORT_443_TCP_PORT=443
KUBERNETES_PORT_443_TCP_PROTO=tcp
KUBERNETES_SERVICE_HOST=10.152.183.1
KUBERNETES_SERVICE_PORT=443
KUBERNETES_SERVICE_PORT_HTTPS=443
MODELOPERATOR_PORT=tcp://10.152.183.153:17071
MODELOPERATOR_PORT_17071_TCP=tcp://10.152.183.153:17071
MODELOPERATOR_PORT_17071_TCP_ADDR=10.152.183.153
MODELOPERATOR_PORT_17071_TCP_PORT=17071
MODELOPERATOR_PORT_17071_TCP_PROTO=tcp
MODELOPERATOR_SERVICE_HOST=10.152.183.153
MODELOPERATOR_SERVICE_PORT=17071
PATH=/usr/share/grafana/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
PEBBLE_SOCKET=/charm/container/pebble.socket
TZ=UTC
```

Where are `GF_SERVER_SERVE_FROM_SUBPATH` and `GF_SERVER_ROOT_URL`?

**Comment by PietroPasotti** [comment]
I also had similar issues last week; thought it was related to https://github.com/canonical/pebble/pull/96 
It was fixed for me after updating to juju 2.9.23.

**Comment by rbarry82** [comment]
Testing with 2.9.24 is ok for me also. Closing

**Comment by benhoyt** [comment]
Yes, this was fixed in https://github.com/canonical/pebble/commit/3a48156fcbf025462e4d063a9939307528017d7f, which was included in Juju 2.9.23 (and hence is also fixed in 2.9.24). If you're installing from the Juju snap, you'll need to grab `latest/candidate`. Hopefully stable will be updated to 2.9.24 soon.
